lib/sdn_epc/ofsh_call.ex

```diff
@@ -5,31 +5,51 @@ defmodule SdnEpc.OfshCall do
   # :ofs_handler callbacks
 
   def init(_mode, _ip, datapath_id, _features, _version, _connection, _opts) do
-    {:ok, ofpc_sup} = start_ofp_channel_sup()
-    open_ofp_channel(ofpc_sup)
-    SdnEpc.Forwarder.save_datapath_id(datapath_id)
-    SdnEpc.Forwarder.subscribe_messages_from_switch(datapath_id,
-      [:packet_in, :features_reply, :port_desc_reply])
+    establish_connection_with_controller(datapath_id)
+    subscribe_for_switch_messages(datapath_id)
     {:ok, datapath_id}
   end
 
-  def handle_message(msg, _datapath_id)  do
+  def handle_message(msg, datapath_id)  do
     Logger.debug("Message received from switch")
-    switch_id = Application.get_env(:sdn_epc, :switch_id)
-    SdnEpc.Forwarder.send_msg_to_controller(switch_id, msg)
+    switch_id = convert_mac_addr_to_int(datapath_id)
```

Comment:
Why this conversion? Cannot you identify the switch by the datapath_id as it is?